src/Npgsql/NpgsqlConnection.cs

```diff
@@ -182,6 +183,34 @@ void GetPoolAndSettings()
             _pool = PoolManager.GetOrAdd(_connectionString, _pool);
         }
 
+        NpgsqlConnectionStringBuilder ParseConnectionString(string connectionString)
+        {
+            return Uri.TryCreate(connectionString, UriKind.Absolute, out var uri)
+                ? ParseSettingsFromUri(uri)
+                : new NpgsqlConnectionStringBuilder(_connectionString);
+        }
+
+        NpgsqlConnectionStringBuilder ParseSettingsFromUri(Uri uri)
+        {
+            var builder = new NpgsqlConnectionStringBuilder();
+
+            var userInfo = uri.UserInfo?.Split(':');
```

Comment:
I think we'll also need to [unescape](https://docs.microsoft.com/dotnet/api/system.uri.unescapedatastring) some values here.